modules/ags/src/main/scala/lucuma/ags/AgsAnalysis.scala

```diff
@@ -158,22 +157,32 @@ object AgsAnalysis {
     /**
      * This method will sort the analysis for quality and internally for positions that give the
      * lowest vignetting.
+     *
+     * Non usable positions wii be discarded.
      */
-    def sortPositions(positions: NonEmptyList[AgsPosition]): List[AgsAnalysis] = {
-      val (usable, nonUsable)                = results.partition(_.isUsable)
+    def sortUsablePositions(positions: NonEmptyList[AgsPosition]): List[AgsAnalysis] = {
       val usablePerTarget: List[AgsAnalysis] =
-        usable
+        results
           .groupBy(_.target.id)
-          .map { (_, analyses) =>
-            analyses
-              .collect { case u: Usable =>
-                u
-              }
-              .reduce((a, b) => a.copy(vignetting = a.vignetting.concatNel(b.vignetting)))
-              .sortedVignetting
+          .flatMap { (_, analyses) =>
+            val usable = analyses.collect {
+              case r if r.isUsable => r
+            }
+            // If there is at least a single usable result, we can sort by vignetting
+            // and discard the non usable results
+            if (usable.nonEmpty) {
+              List(
+                usable
+                  .collect { case u: Usable =>
+                    u
+                  }
```

Comment:
Or maybe this is the `collect` we want previously?

Reply:
Right, they could be merged, `collect` is useful as it lets me get a `List[Usable]` rather than a `List[AgsAnalysis]` but it seems we are iterating twice